autoreject/tests/test_autoreject.py

```diff
@@ -195,6 +201,30 @@ def test_autoreject():
     if not np.isscalar(is_same):
         is_same = np.isscalar(is_same)
     assert_true(not is_same)
+
+    # test that transform ignores bad channels
+    picks_without_bads = mne.pick_types(
+        epochs_with_bads.info, meg='mag', eeg=True, stim=False, eog=False,
+        include=[], exclude='bads')
+    ar_bads = LocalAutoRejectCV(cv=3, picks=picks_without_bads,
```

Comment:
this test is "trivial" if the picks is working, no? That is to say, if I pass any arbitrary integer list (with or without bad channels), it should just respect the `picks` that the user provides. I would be surprised if this failed on master. The case of `picks=None` is different as it is modified inside `autoreject` ...

Reply:
It fails on master because the EEG interpolation code didn't respect the picks. The previous code reset the user specified bads to good channels to be used for interpolation instead of excluding them from the interpolation process. This test directly tests for this bug by checking that the presence of bad EEG channels in the data doesn't change the outcome of autoreject on the good channels. So I don't think it's necessarily trivial given that the picks the user provided weren't being fully respected before. I agree that it'd be better to check the handling of picks=None in addition so I updated the test to check this. Does this seem reasonable?